precompiles/act/act.go

```diff
@@ -0,0 +1,162 @@
+package act
+
+import (
+	"embed"
+	"github.com/warden-protocol/wardenprotocol/precompiles/common"
+
+	"cosmossdk.io/log"
+	storetypes "cosmossdk.io/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	ethcmn "github.com/ethereum/go-ethereum/common"
+	evmoscmn "github.com/evmos/evmos/v20/precompiles/common"
+	"github.com/evmos/evmos/v20/x/evm/core/vm"
+
+	actmodulekeeper "github.com/warden-protocol/wardenprotocol/warden/x/act/keeper"
+)
+
+var _ vm.PrecompiledContract = &Precompile{}
+
+const PrecompileAddress = "0x0000000000000000000000000000000000000901"
+
+// Embed abi json file to the executable binary. Needed when importing as dependency.
+//
+//go:embed abi.json
+var f embed.FS
+
+// Precompile defines the precompiled contract for x/act.
+type Precompile struct {
+	evmoscmn.Precompile
+	actmodulekeeper actmodulekeeper.Keeper
+	eventsRegistry  *common.EthEventsRegistry
+}
+
+// LoadABI loads the x/act ABI from the embedded abi.json file
+// for the x/act precompile.
+func LoadABI() (abi.ABI, error) {
+	return evmoscmn.LoadABI(f, "abi.json")
+}
+
+func NewPrecompile(actkeeper actmodulekeeper.Keeper, er *common.EthEventsRegistry) (*Precompile, error) {
+	abi, err := LoadABI()
+	if err != nil {
+		return nil, err
+	}
+
+	p := Precompile{
+		Precompile: evmoscmn.Precompile{
+			ABI:                  abi,
+			KvGasConfig:          storetypes.KVGasConfig(),
+			TransientKVGasConfig: storetypes.TransientGasConfig(),
+		},
+		actmodulekeeper: actkeeper,
+		eventsRegistry:  er,
+	}
+
```

Comment:
```suggestion
	p.SetAddress(p.Address())

```

I don't know exactly how and when this is used inside evmos code, but we should set it, just in case.